wagtail/users/utils.py

```diff
@@ -25,24 +27,37 @@ def user_can_delete_user(current_user, user_to_delete):
     return True
 
 
-def get_gravatar_url(email, size=50):
-    default = "mm"
-    size = (
-        int(size) * 2
-    )  # requested at retina size by default and scaled down at point of use with css
+def get_gravatar_url(email, size=50, default_params={}):
+    DEFAULT_IMAGE = "mp"
+    DEFAULT_GRAVATAR_PROVIDER_URL = "www.gravatar.com/avatar"
+
+    params = {
+        "s": int(size)
+        * 2,  # requested at retina size by default and scaled down at point of use with css
+        "d": DEFAULT_IMAGE,
+    }
+
     gravatar_provider_url = getattr(
-        settings, "WAGTAIL_GRAVATAR_PROVIDER_URL", "//www.gravatar.com/avatar"
+        settings, "WAGTAIL_GRAVATAR_PROVIDER_URL", DEFAULT_GRAVATAR_PROVIDER_URL
     )
 
     if (not email) or (gravatar_provider_url is None):
         return None
 
-    email_bytes = email.lower().encode("utf-8")
-    hash = safe_md5(email_bytes, usedforsecurity=False).hexdigest()
-    gravatar_url = "{gravatar_provider_url}/{hash}?{params}".format(
-        gravatar_provider_url=gravatar_provider_url.rstrip("/"),
-        hash=hash,
-        params=urlencode({"s": size, "d": default}),
+    parsed_url = urlparse(gravatar_provider_url.rstrip("/"))
+
+    if parsed_url.query:
+        params.update(
+            **{key: value[-1] for key, value in parse_qs(parsed_url.query).items()},
+        )
+    params.update(**default_params)
+
+    email_hash = safe_md5(
+        email.lower().encode("utf-8"), usedforsecurity=False
+    ).hexdigest()
+
+    gravatar_url = (
+        f"//{parsed_url.netloc}{parsed_url.path}/{email_hash}?{urlencode(params)}"
```

Comment:
```suggestion
        f"//{parsed_url.netloc}{parsed_url.path}/{email_hash}?{urlencode(params, doseq=True)}"
```

This should encode the URL params back to duplicate values instead of formatting an array.
https://docs.djangoproject.com/en/5.0/ref/utils/#django.utils.http.urlencode

This is based on the Python urlencode which has the same kwarg
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode

Also, I would recommend instead of building up the URL manually like this, you try to use the urljoin function.

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

or maybe `unsplit`.